source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -72,8 +212,26 @@ class AuthenticatorImpl : public Logger::Loggable<Logger::Id::filter>,
   std::string uri_;
   // The pending remote request so it can be canceled.
   Http::AsyncClient::Request* request_{};
+
+  std::vector<JwtMatcherConstPtr> matchers_;
+
+  bool okToBypass_;
 };
 
+AuthenticatorImpl::AuthenticatorImpl(FilterConfigSharedPtr config)
+    : config_(config), okToBypass_(false) {
+  for (const auto& rule : config_->getProtoConfig().rules()) {
```

Comment:
matchers_  should be created per filter config,   not per request.   Here Authenticator is created per-request.

You can move them here <https://github.com/envoyproxy/envoy/blob/master/source/extensions/filters/http/jwt_authn/filter_config.h#L59>